internal/provider/provider.go

```diff
@@ -117,13 +127,31 @@ type metaData struct {
 
 func providerAuthenticate(ctx context.Context, d *schema.ResourceData, md *metaData) error {
 	var credentials map[string]interface{}
+	providerScope := DEFAULT_PROVIDER_SCOPE
+	amClient := authmethods.NewClient(md.client)
 
 	authMethodId, authMethodIdOk := d.GetOk("auth_method_id")
+
+	scopeId, scopeIdOk := d.GetOk("scope_id")
+	if scopeIdOk {
+		providerScope = scopeId.(string)
+	}
+
 	recoveryKmsHcl, recoveryKmsHclOk := d.GetOk("recovery_kms_hcl")
 	if token, ok := d.GetOk("token"); ok {
 		md.client.SetToken(token.(string))
 	}
 
+	// If auth_method_id is not set, get the default auth method ID for the given scope ID
```

Comment:
this might be easier to read if it's something like:
```
	var providerScope string
	scopeId, scopeIdOk := d.GetOk("scope_id")
	switch {
	case scopeIdOk:
		providerScope = scopeId.(string)
	default:
		providerScope = DEFAULT_PROVIDER_SCOPE
	}
```